src/Shell/Command/GitDriver.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CloudPatches\Shell\Command;
+
+use Magento\CloudPatches\Patch\PatchCommandException;
+use Magento\CloudPatches\Shell\ProcessFactory;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+
+/**
+ * GIT patch driver
+ */
+class GitDriver implements DriverInterface
+{
+    /**
+     * @var ProcessFactory
+     */
+    private $processFactory;
+
+    /**
+     * @param ProcessFactory $processFactory
+     */
+    public function __construct(
+        ProcessFactory $processFactory
+    ) {
+        $this->processFactory = $processFactory;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function apply(string $patch)
+    {
+        try {
+            $this->processFactory->create(['git', 'apply'], $patch)
+                ->mustRun();
+        } catch (ProcessFailedException $exception) {
+            throw new PatchCommandException('Failed to apply patch', $exception->getCode(), $exception);
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function revert(string $patch)
+    {
+        try {
+            $this->processFactory->create(['git', 'apply', '--reverse'], $patch)
+                ->mustRun();
+        } catch (ProcessFailedException $exception) {
+            throw new PatchCommandException('Failed to revert patch', $exception->getCode(), $exception);
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function applyCheck(string $patch)
+    {
+        try {
+            $this->processFactory->create(['git', 'apply', '--check'], $patch)
+                ->mustRun();
+        } catch (ProcessFailedException $exception) {
+            throw new PatchCommandException('Patch cannot be applied', $exception->getCode(), $exception);
```

Comment:
This error message contains Git error output that is parsed further https://github.com/magento-mpi/magento-cloud-patches/blob/72451669ad3d6b7b47b15aeeda0067c24cd717a1/src/Command/Process/Action/ApplyOptionalAction.php#L105 
https://github.com/magento-mpi/magento-cloud-patches/blob/72451669ad3d6b7b47b15aeeda0067c24cd717a1/src/Command/Process/ApplyLocal.php#L98

Actually here
https://github.com/magento-mpi/magento-cloud-patches/blob/72451669ad3d6b7b47b15aeeda0067c24cd717a1/src/Command/Process/Renderer.php#L151

We need to show the customer error details